diff --git a/node_modules/react-native-screens/android/CMakeLists.txt b/node_modules/react-native-screens/android/CMakeLists.txt
index 86ec079..928d0c7 100644
--- a/node_modules/react-native-screens/android/CMakeLists.txt
+++ b/node_modules/react-native-screens/android/CMakeLists.txt
@@ -43,6 +43,7 @@ if(${RNS_NEW_ARCH_ENABLED})
     
     if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
         target_link_libraries(rnscreens
+            c++_shared
             ReactAndroid::reactnative
             ReactAndroid::jsi
             fbjni::fbjni
@@ -51,6 +52,7 @@ if(${RNS_NEW_ARCH_ENABLED})
     else()
         target_link_libraries(
             rnscreens
+                c++_shared
                 ReactAndroid::jsi
                 ReactAndroid::react_nativemodule_core
                 ReactAndroid::react_utils
@@ -74,6 +76,7 @@ if(${RNS_NEW_ARCH_ENABLED})
     endif()
 else()
     target_link_libraries(rnscreens
+        c++_shared
         ReactAndroid::jsi
         android
     )
diff --git a/node_modules/react-native-screens/android/build/generated/source/codegen/jni/CMakeLists.txt b/node_modules/react-native-screens/android/build/generated/source/codegen/jni/CMakeLists.txt
new file mode 100644
index 0000000..5f34bea
--- /dev/null
+++ b/node_modules/react-native-screens/android/build/generated/source/codegen/jni/CMakeLists.txt
@@ -0,0 +1,29 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+cmake_minimum_required(VERSION 3.13)
+set(CMAKE_VERBOSE_MAKEFILE on)
+
+file(GLOB react_codegen_SRCS CONFIGURE_DEPENDS *.cpp react/renderer/components/rnscreens/*.cpp)
+
+add_library(
+  react_codegen_rnscreens
+  OBJECT
+  ${react_codegen_SRCS}
+)
+
+target_include_directories(react_codegen_rnscreens PUBLIC . react/renderer/components/rnscreens)
+
+target_link_libraries(
+  react_codegen_rnscreens
+  c++_shared
+  fbjni
+  jsi
+  # We need to link different libraries based on whether we are building rncore or not, that's necessary
+  # because we want to break a circular dependency between react_codegen_rncore and reactnative
+  reactnative
+)
+
+target_compile_reactnative_options(react_codegen_rnscreens PRIVATE)
diff --git a/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
index 1b26dac..30a708b 100644
--- a/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-screens/android/src/main/jni/CMakeLists.txt
@@ -42,6 +42,7 @@ target_include_directories(
 if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
   target_link_libraries(
     ${LIB_TARGET_NAME}
+    c++_shared
     ReactAndroid::reactnative
     ReactAndroid::jsi
     fbjni::fbjni
@@ -49,6 +50,7 @@ if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
 else()
   target_link_libraries(
     ${LIB_TARGET_NAME}
+    c++_shared
     fbjni
     folly_runtime
     glog